### ISSUE TITLE
_mm_malloc and _mm_free

### ISSUE THREAD

**Issue by jeffhammond** [simd-everywhere/simde#940]
Many codes that use SSE and friends also use `_mm_{malloc,realloc,free}`.

SSE2Neon supports these (https://github.com/DLTcollab/sse2neon/pull/25/files).

The implementation is simple and I will try to contribute if no one else does it first.

Thanks.

**Comment by thomasdwu** [comment]
I was wondering whether there was any progress on this issue.

**Comment by jeffhammond** [comment]
Yes. https://stackoverflow.com/q/32612881/2189128 has some context.

**Comment by jeffhammond** [comment]
You'll have to replace _mm_free as well, but just with free().

**Comment by thomasdwu** [comment]
Got it, thanks.  I also see that posix_memalign() takes its argument in terms of bits, whereas _mm_malloc() takes it in terms of bytes.

**Comment by jeffhammond** [comment]
The alignment argument is in bytes from what I can tell.

> The value of alignment shall be a power of two multiple of sizeof(void *).

https://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_memalign.html

**Comment by thomasdwu** [comment]
Yes, you're right.  Somehow, I was mistakenly thinking that addresses were in terms of bits.
